swig/python/gdal-utils/osgeo_utils/gdal_retile.py

```diff
@@ -505,19 +523,21 @@ def createPyramidTile(
             sy,
         ]
     )
-
-    if OGRDS is not None:
+    # if -resume flag and the tile is present, add it to the index and exit function
+    if OGRDS is not None and feature_only:
```

Comment:
As far as I can see OGRDS cannot be None:
```suggestion
    if feature_only:
```